### ISSUE TITLE
snowflake_tag_association does not work on SCHEMA object_types

### ISSUE THREAD

**Issue by ericmjalbert** [Snowflake-Labs/terraform-provider-snowflake#1301]
**Provider Version**

0.48.0

**Terraform Version**

v1.2.3

**Describe the bug**

Using the `snowflake_tag_association` on a schema does not work. The generated SQL query that it runs errors and thus the terraform project can never successfully apply.

The SQL that is run on Snowflake looks like:
```
ALTER SCHEMA "STORAGE".."TEST" SET TAG "STORAGE"."PUBLIC"."EXAMPLE_TAG" = 'TAG_VALUE'
```

which noticeably has a typo in the way the object name is generated (it includes an extra `.` in the `"STORAGE".."TEST"` identifier).

**Expected behavior**

Whenever using the `snowflake_tag_association` with an `object_type = "SCHEMA"` the provider should change the way it generates the SQL query so that it doesn't include an empty value for the `object_identifier.schema` part. For example, it should look like:

```
ALTER SCHEMA "STORAGE"."TEST" SET TAG "STORAGE"."PUBLIC"."EXAMPLE_TAG" = 'TAG_VALUE'
```

**Code samples and commands**

```
resource "snowflake_tag" "example_tag" {
  database = "STORAGE"
  name     = "EXAMPLE_TAG"
  schema   = "PUBLIC"

  allowed_values = [""]
}

resource "snowflake_tag_association" "schema" {
  object_identifier {
    database = "STORAGE"
    name     = "TEST"
  }

  object_type = "SCHEMA"
  tag_id      = snowflake_tag.example_tag.id
  tag_value   = "TAG_VALUE"
}
```

**Additional context**

**Comment by systematics78** [comment]
00:58:12  │ Error: Request cancelled
00:58:12  │ 
00:58:12  │   with snowflake_tag_association.CROSS_REGION_DATA_SHARING_UC_tag_assosiation["CLINICAL_OPERATIONS_CONSOL_LAUNCH360.DRUG_DEVELOPMENT_REFINEMENT.CLINICAL_OPERATIONS_CONSOL.STUDY_EVENT_METRIC.F1_PRD_SNO_US_L360_DATA_ENGINEER"],
00:58:12  │   on cross_region_data_sharing.tf line 430, in resource "snowflake_tag_association" "CROSS_REGION_DATA_SHARING_UC_tag_assosiation":
00:58:12  │  430: resource "snowflake_tag_association" "CROSS_REGION_DATA_SHARING_UC_tag_assosiation" {
00:58:12  │ 
00:58:12  │ The plugin.(*GRPCProvider).UpgradeResourceState request was cancelled.

**Comment by systematics78** [comment]
Also I'm getting provided above error - request canceled,

**Comment by systematics78** [comment]
23:22:19  Stack trace from the terraform-provider-snowflake_v0.49.0 plugin:
23:22:19  
23:22:19  panic: runtime error: index out of range [0] with length 0
23:22:19  
23:22:19  goroutine 2960 [running]:
23:22:19  github.com/Snowflake-Labs/terraform-provider-snowflake/pkg/resources.ReadTagAssociation(0xc0004b5a80, {0x1270b60?, 0xc0001e09c0})
23:22:19  	github.com/Snowflake-Labs/terraform-provider-snowflake/pkg/resources/tag_association.go:163 +0x4d7
23:22:19  github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).read(0x14c9420?, {0x14c9420?, 0xc001024ae0?}, 0xd?, {0x1270b60?, 0xc0001e09c0?})
23:22:19  	github.com/hashicorp/terraform-plugin-sdk/v2@v2.24.0/helper/schema/resource.go:712 +0x178
23:22:19  github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).RefreshWithoutUpgrade(0xc00078f340, {0x14c9420, 0xc001024ae0}, 0xc000eaf6c0, {0x1270b60, 0xc0001e09c0})
23:22:19  	github.com/hashicorp/terraform-plugin-sdk/v2@v2.24.0/helper/schema/resource.go:1015 +0x585
23:22:19  github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*GRPCProviderServer).ReadResource(0xc0004b1e18, {0x14c9420?, 0xc001024900?}, 0xc000ad5340)
23:22:19  	github.com/hashicorp/terraform-plugin-sdk/v2@v2.24.0/helper/schema/grpc_provider.go:613 +0x4a5
23:22:19  github.com/hashicorp/terraform-plugin-go/tfprotov5/tf5server.(*server).ReadResource(0xc000453720, {0x14c9420?, 0xc001024330?}, 0xc0004d7500)
23:22:19  	github.com/hashicorp/terraform-plugin-go@v0.14.0/tfprotov5/tf5server/server.go:748 +0x4b1
23:22:19  github.com/hashicorp/terraform-plugin-go/tfprotov5/internal/tfplugin5._Provider_ReadResource_Handler({0x123d9e0?, 0xc000453720}, {0x14c9420, 0xc001024330}, 0xc000396770, 0x0)
23:22:19  	github.com/hashicorp/terraform-plugin-go@v0.14.0/tfprotov5/internal/tfplugin5/tfplugin5_grpc.pb.go:349 +0x170
23:22:19  google.golang.org/grpc.(*Server).processUnaryRPC(0xc0005bc000, {0x14cede0, 0xc000007860}, 0xc000e018c0, 0xc000780870, 0x1c33290, 0x0)
23:22:19  	google.golang.org/grpc@v1.48.0/server.go:1295 +0xb2b
23:22:19  google.golang.org/grpc.(*Server).handleStream(0xc0005bc000, {0x14cede0, 0xc000007860}, 0xc000e018c0, 0x0)
23:22:19  	google.golang.org/grpc@v1.48.0/server.go:1636 +0xa2f
23:22:19  google.golang.org/grpc.(*Server).serveStreams.func1.2()
23:22:19  	google.golang.org/grpc@v1.48.0/server.go:932 +0x98
23:22:19  created by google.golang.org/grpc.(*Server).serveStreams.func1
23:22:19  	google.golang.org/grpc@v1.48.0/server.go:930 +0x28a
23:22:19  
23:22:19  Error: The terraform-provider-snowflake_v0.49.0 plugin crashed!
23:22:19  
23:22:19  This is always indicative of a bug within the plugin. It would be immensely
23:22:19  helpful if you could report the crash with the plugin's maintainers so that it
23:22:19  can be fixed. The output above should help diagnose the issue.

**Comment by systematics78** [comment]
@mlorek I do not think that it is got fixed. I'm still getting error which I have described above

**Comment by systematics78** [comment]
By the way - I'm applying tags to ROLES

**Comment by mlorek** [comment]
not merged yet

**Comment by ericmjalbert** [comment]
I'm going to close this because the original issue is now resolved since the above commit got merged in version `0.51.0`. I just ran it today on the latest release (`0.55.1`) and it correctly applies the TAGS to schemas. Below is the SQL i saw in the activity logs on my snowflake account:

```
ALTER SCHEMA "STORAGE"."TEST" SET TAG "TAGS"."PUBLIC"."EXAMPLE_TAG" = 'TAG_VALUE'
```
